File: usecase/src/main/java/br/org/restaurantepopular/usecase/product_type/DeleteProductTypeUseCase.java
package br.org.restaurantepopular.usecase.product_type;

import br.org.restaurantepopular.usecase.core.IUseCase;

import java.util.Optional;

public class DeleteProductTypeUseCase implements IUseCase<String, Optional> {

    private final IProductTypeRepository productTypeRepository;

    public DeleteProductTypeUseCase(IProductTypeRepository productTypeRepository) {
        this.productTypeRepository = productTypeRepository;
    }

    @Override
    public Optional execute(String s) {
        productTypeRepository.delete(s);
        return Optional.empty();
    }
}

File: usecase/src/main/java/br/org/restaurantepopular/usecase/product_type/DeleteProductTypeUseCase.kt
package br.org.restaurantepopular.usecase.product_type

import br.org.restaurantepopular.usecase.core.IUseCase

class DeleteProductTypeUseCase(
    private val productTypeRepository: IProductTypeRepository
) : IUseCase<String, Unit> {

    override fun execute(s: String) {
        productTypeRepository.delete(s)
    }
}
